### ISSUE TITLE
Bor increasing memory usage

### ISSUE THREAD

**Issue by aaa8212** [maticnetwork/bor#1145]
#### **System information**

Bor client version: v1.2.1 and v1.2.3

Heimdall client version: v1.0.3

OS & Version: Linux

Environment: Polygon Mainnet

Type of node: Full node

Additional Information: /

#### **Overview of the problem**

Bor memory usage is increasing until machine crashes / OOM happens. The rate is quite high, in 14 hours it uses about 128 GB of RAM.

What is interesting is that version `v1.2.1` was running for about 3 few weeks without problems (normal memory usage - did not increase over time), then two weeks ago it started consuming memory very steeply. We do not know what happened then / what has changed.

Upgrading to `v1.2.3` didn't help, the symptom is exactly the same.

#### **Reproduction Steps**

Start bor v1.2.1 (as Docker container) and monitor its memory usage.

#### **Logs / Traces / Output / Error Messages**

```
 # date; smem -t -k -P '^bor server' -m
Thu Jan 25 09:41:18 AM CET 2024
Map                                       PIDs   AVGPSS      PSS
[vdso]                                       1        0        0
[vsyscall]                                   1        0        0
[vvar]                                       1        0        0
[stack]                                      1   224.0K   224.0K
[heap]                                       1   712.0K   712.0K
/usr/bin/bor                                 1    38.1M    38.1M
<anonymous>                                  1    13.6G    13.6G
-----------------------------------------------------------------
7                                            7    13.6G    13.6G
# date; smem -t -k -P '^bor server' -m
Thu Jan 25 10:22:44 AM CET 2024
Map                                       PIDs   AVGPSS      PSS
[vdso]                                       1        0        0
[vsyscall]                                   1        0        0
[vvar]                                       1        0        0
[stack]                                      1   224.0K   224.0K
[heap]                                       1   712.0K   712.0K
/usr/bin/bor                                 1    20.9M    20.9M
<anonymous>                                  1    20.9G    20.9G
-----------------------------------------------------------------
7                                            7    20.9G    20.9G
```

![image](https://github.com/maticnetwork/bor/assets/93202194/478a43c4-9fbe-42a1-bdd0-0faa2758eaa7)

**Additional Information**

We tried running `go tool pprof -png http://bor:6060/debug/pprof/heap > mem.png`, the result is below. What is interesting is that memory usage reported this way does *not* show the excessive memory usage, not sure why. This is the result:

![mem](https://github.com/maticnetwork/bor/assets/93202194/9e360b26-1cdd-4e4c-90cc-b8d5f70b1f4e)

Any idea how to solve this? Happy to debug if needed.

**Comment by aaa8212** [comment]
Does someone have an idea how to solve this?

**Comment by temaniarpit27** [comment]
We have tested v1.2.3 thoroughly and we don't see any signs of increasing memory. Neither any partner has reported this.
Can you please check once if you are using the current version?

**Comment by aaa8212** [comment]
We no longer experience this problem, though the reason is unclear. Writing down the process in case it helps someone.

In the mean time we needed to resize partitions on our RAID array. During process (which should be transparent) bor crashed with error message regarding corruption of some old file:
```
Failed to store last block's hash err="leveldb/table: corruption on data-block (pos=865655): checksum mismatch, want=0x0 got=0xedb39123 [file=155847618.ldb]"
```
The file in question was from 15th of January, which is before the memory usage problems and before resizing of partitions.

When we replaced the file from snapshots and restarted bor, the memory usage problem disappeared:

![image](https://github.com/maticnetwork/bor/assets/93202194/3479eab9-6b6e-4072-9746-6958eb786bec)

Feel free to close this issue.
